File: Patterns/Diamond.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int i, j, r, x;
    cout<<"Enter the Number of Rows: ";
    cin>>r;
    x = r-1;
    for(i=1; i<=r; i++)
    {
        for(j=1; j<=x; j++)
            cout<<" ";
        x--;
        for(j=1; j<=(2*i-1); j++)
            cout<<"*";
        cout<<endl;
    }
    x = 1;
    for(i=1; i<=(r-1); i++)
    {
        for(j=1; j<=x; j++)
            cout<<" ";
        x++;
        for(j=1; j<=(2*(r-i)-1); j++)
            cout<<"*";
        cout<<endl;
    }
    cout<<endl;
    return 0;
}

File: Patterns/diamond.py
import java.util.Scanner;
public class Edureka
{
    public static void main(String[] args)
{
    Scanner sc = new Scanner(System.in);
 
    System.out.println("Enter the number of rows: ");
 
    int rows = sc.nextInt();    
    for (int i=1; i<= rows ; i++) { for (int j = rows; j > i ; j--) {
            System.out.print(" ");
        }
        System.out.print("*");
        for (int k = 1; k < 2*(i -1) ;k++) { System.out.print(" "); } if( i==1) { System.out.println(""); } else { System.out.println("*"); } } for (int i=rows-1; i>= 1 ; i--)
        {
        for (int j = rows; j > i ; j--) {
            System.out.print(" ");
        }
        System.out.print("*");
        for (int k = 1; k < 2*(i -1) ;k++) {
            System.out.print(" ");
        }
        if( i==1)
            System.out.println("");
        else
            System.out.println("*");
    }
    sc.close();
}
}
